src/core/trulens/experimental/otel_tracing/core/trace.py

```diff
@@ -608,118 +625,68 @@ def records_of_recording(
-    def _span(self, cls, **kwargs):
+    def _span(self, cls: Type[S], **kwargs) -> ContextManager[S]:
         with self.start_span(cls=cls, **kwargs) as span:
-            token = self.context_cvar.set(span.context)
-
-            try:
+            with python_utils.with_context({self.context_cvar: span.context}):
                 yield span
 
-            finally:
-                self.context_cvar.reset(token)
-
     @contextlib.asynccontextmanager
-    async def _aspan(self, cls, **kwargs):
+    async def _aspan(self, cls: Type[S], **kwargs) -> ContextManager[S]:
         async with self.start_span(cls=cls, **kwargs) as span:
-            token = self.context_cvar.set(span.context)
-
-            try:
+            async with python_utils.awith_context({
+                self.context_cvar: span.context
+            }):
                 yield span
 
-            finally:
-                self.context_cvar.reset(token)
-
     # context manager
-    def recording(self):
+    def recording(self) -> ContextManager[PhantomSpanRecordingContext]:
         return self._span(
             name="trulens.recording", cls=PhantomSpanRecordingContext
         )
 
     # context manager
-    def method(self, method_name: str):
+    def method(self, method_name: str) -> ContextManager[LiveSpanCall]:
         return self._span(name="trulens.call." + method_name, cls=LiveSpanCall)
 
     # context manager
-    def cost(self, method_name: str, cost: Optional[base_schema.Cost] = None):
+    def cost(
+        self, method_name: str, cost: Optional[base_schema.Cost] = None
+    ) -> ContextManager[LiveSpanCallWithCost]:
         return self._span(
             name="trulens.call." + method_name,
             cls=LiveSpanCallWithCost,
             cost=cost,
         )
 
     # context manager
-    def phantom(self):
+    def phantom(self) -> ContextManager[PhantomSpan]:
         return self._span(name="trulens.phantom", cls=PhantomSpan)
 
     # context manager
-    async def arecording(self):
+    async def arecording(self) -> ContextManager[PhantomSpanRecordingContext]:
         return self._aspan(
             name="trulens.recording", cls=PhantomSpanRecordingContext
         )
 
     # context manager
-    async def amethod(self, method_name: str):
+    async def amethod(self, method_name: str) -> ContextManager[LiveSpanCall]:
         return self._aspan(name="trulens.call." + method_name, cls=LiveSpanCall)
 
     # context manager
     async def acost(
         self, method_name: str, cost: Optional[base_schema.Cost] = None
-    ):
+    ) -> ContextManager[LiveSpanCallWithCost]:
         return self._aspan(
             name="trulens.call." + method_name,
             cls=LiveSpanCallWithCost,
             cost=cost,
         )
 
     # context manager
-    async def aphantom(self):
+    async def aphantom(self) -> ContextManager[PhantomSpan]:
         return self._aspan(name="trulens.phantom", cls=PhantomSpan)
 
 
-class NullTracer(Tracer):
```

Comment:
can you add a comment about why we don't need NullTracer in TruLens at the moment?

Reply:
added as a TODO in class Tracer